src/components/webview/CampaignManagerProvider.ts

```diff
@@ -0,0 +1,355 @@
+import {
+	WebviewViewProvider,
+	WebviewView,
+	Uri,
+	ExtensionContext,
+	workspace,
+	commands,
+} from 'vscode';
+import { Message, MessageBus, MessageKind } from '../messageBus';
+import {
+	TreeNode,
+	View,
+	WebviewMessage,
+	WebviewResponse,
+} from './webviewEvents';
+import { WebviewResolver } from './WebviewResolver';
+import {
+	CaseElement,
+	Element,
+	ElementHash,
+	FileElement,
+} from '../../elements/types';
+import { Job, JobHash, JobKind } from '../../jobs/types';
+import { debounce, getElementIconBaseName } from '../../utilities';
+import { JobManager } from '../jobManager';
+import { CaseWithJobHashes } from '../../cases/types';
+import {
+	buildJobElement,
+	compareJobElements,
+} from '../../elements/buildJobElement';
+import {
+	buildFileElement,
+	compareFileElements,
+} from '../../elements/buildFileElement';
+import {
+	buildCaseElement,
+	compareCaseElements,
+} from '../../elements/buildCaseElement';
+import { CaseManager } from '../../cases/caseManager';
+import { SourceControlService } from '../sourceControl';
+
+export class CampaignManagerProvider implements WebviewViewProvider {
+	__view: WebviewView | null = null;
+	__extensionPath: Uri;
+	__webviewResolver: WebviewResolver | null = null;
+	__elementMap = new Map<ElementHash, Element>();
+	__unsavedChanges = false;
+
+	constructor(
+		context: ExtensionContext,
+		private readonly __messageBus: MessageBus,
+		private readonly __jobManager: JobManager,
+		private readonly __caseManager: CaseManager,
+		private readonly __sourceControl: SourceControlService,
+	) {
+		this.__extensionPath = context.extensionUri;
+
+		this.__webviewResolver = new WebviewResolver(this.__extensionPath);
+	}
+
+	refresh(): void {
+		if (!this.__view) {
+			return;
+		}
+
+		this.__webviewResolver?.resolveWebview(
+			this.__view.webview,
+			'campaignManager',
+			'{}',
+		);
+	}
+
+	resolveWebviewView(webviewView: WebviewView): void | Thenable<void> {
+		if (!webviewView.webview) {
+			return;
+		}
+
+		this.__webviewResolver?.resolveWebview(
+			webviewView.webview,
+			'campaignManager',
+			'{}',
+		);
+		this.__view = webviewView;
+
+		this.__view.onDidChangeVisibility(() => {
+			this.__onUpdateElementsMessage();
+		});
+
+		this.__attachExtensionEventListeners();
+		this.__attachWebviewEventListeners();
+	}
+
+	public setView(data: View) {
+		this.__postMessage({
+			kind: 'webview.global.setView',
+			value: data,
+		});
+	}
+
+	private __postMessage(message: WebviewMessage) {
+		if (!this.__view) {
+			return;
+		}
+
+		this.__view.webview.postMessage(message);
```

Comment:
`this.__view?.webview.postMessage(message)`
then the lines 101-103 are not needed